litgpt/adapter.py

```diff
@@ -91,30 +62,10 @@ def _init_weights(self, module: nn.Module) -> None:
 
 
 class Block(BaseBlock):
-    """The implementation is identical to `litgpt.model.Block` with the exception that
-    we replace the attention layer where adaption is implemented."""
-
     def __init__(self, config: Config, block_idx: int) -> None:
-        # Skip the parent class __init__ altogether and replace it to avoid useless allocations
-        nn.Module.__init__(self)
-        if not config.parallel_residual and config.shared_attention_norm:
-            raise NotImplementedError(
-                "No checkpoint amongst the ones we support uses this configuration:"
-
-                " non-parallel residual and shared attention norm."
-            )
-        self.norm_1 = config.norm_class(config.n_embd, eps=config.norm_eps)
+        super().__init__(config, block_idx, skip_attn_mlp=True)
         self.attn = CausalSelfAttention(config, block_idx)
-        self.post_attention_norm = (
-            config.norm_class(config.n_embd, eps=config.norm_eps) if config.post_attention_norm else nn.Identity()
-        )
-        self.norm_2 = None if config.shared_attention_norm else config.norm_class(config.n_embd, eps=config.norm_eps)
         self.mlp = config.mlp_class(config)
```

Comment:
Oh, now I see.
You provide `skip_attn_mlp=True`, so in the base class self.attn and self.mlp are assigned to None.
I'm ok with reassigning it here.
Otherwise it might look a bit confusing, why a child class dictates the implementation of the base class.

I think it's better to drop `skip_attn_mlp` at all.
Yes, it will lead to a bit longer initialization, but the code in the base class will be a bit cleaner.

Reply:
OK, makes sense